core/src/main/java/jenkins/views/JenkinsHeader.java

```diff
@@ -0,0 +1,38 @@
+package jenkins.views;
+
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import hudson.Extension;
+import hudson.ExtensionList;
+
+@Extension(ordinal = Integer.MIN_VALUE)
+public class JenkinsHeader implements Header {
+    
+    private static Logger LOGGER = Logger.getLogger(JenkinsHeader.class.getName());
+
+    @Override
+    public boolean isHeaderEnabled() {
+        return true;
+    }
+    
+    @Restricted(NoExternalUse.class)
+    @CheckForNull
+    public static Header get() {
+        List<Header> headers = ExtensionList.lookup(Header.class).stream().filter(header -> header.isHeaderEnabled()).collect(Collectors.toList());
+        if (headers.size() > 0) {
+            if (headers.size() > 1) {
+                LOGGER.warning("More than one configured header. This should not happen. Serving the Jenkins default header and please review");
```

Comment:
how does the user review? is this logging useful to an administrator?

Reply:
also a bit confused isn't the point of this that you want to ship 2 headers, the jenkins one and the cloudbees one, and you want yours to have a higher priority

Reply:
> isn't the point of this that you want to ship 2 headers, the jenkins one and the cloudbees one, and you want yours to have a higher priority

JEP proposes to have just one header loaded in our Jenkins instance. IOW proposal is about to make it feasible to have enough customization techniques from Jenkins to create headers, but we want to limit the number of active implementations of that header to one on runtime. Does it make sense to have more than one header at the same time active in the same Jenkins instance? I do not think so. Do we have the capabilities to limit the headers active? Yes, if header is an extension point we can use an [ExtensionFilter#allows](https://javadoc.jenkins.io/jenkins/ExtensionFilter.html#allows-java.lang.Class-hudson.ExtensionComponent-) method for that purpose.

> how does the user review? is this logging useful to an administrator?
This is shown in the Jenkins log if more than one headers are available. I think it is useful for admins to know that there exist two or more headers active.

Reply:
Why not use a similar aproach to what is currently done for `UpdateCenter` or `PluginManager` (with an override class  [defined](https://github.com/jenkinsci/jenkins/blob/57737eec99d13341b986aeb6db02dda17d2aaf48/core/src/main/java/hudson/PluginManager.java#L185) through a system property?) As you said, it makes no sense to have multiple headers enabled at one point in time, and this customization is likely desired at OEM level (CloudBees shipping its own war, or an enterprise providing its own header implementation through system property).

Reply:
I did not consider that approach. BTW I would prefer doing this transparently without the need of setting extra/additional system properties.

Reply:
All comments addressed. Finally no need to use ExtensionFilter, also getting only the one with top priority and issues reported on warnings and admin monitor. Regarding the alternative approach, as mentioned, preferred not to set up extra/additional system properties.